source/features/ci-link.tsx

```diff
@@ -1,37 +1,60 @@
 import './ci-link.css';
 import select from 'select-dom';
-import onetime from 'onetime';
 import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 import fetchDom from '../helpers/fetch-dom';
-import {buildRepoURL} from '../github-helpers';
+import getDefaultBranch from '../github-helpers/get-default-branch';
+import {buildRepoURL, getCurrentCommittish} from '../github-helpers';
 
-// Look for the CI icon in the latest 2 days of commits #2990
-const getIcon = onetime(async () => fetchDom(
-	buildRepoURL('commits'), [
-		'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses', // TODO[2022-04-29]: GHE #4294
-		'.TimelineItem--condensed:nth-of-type(-n+2) batch-deferred-content[data-url$="checks-statuses-rollups"]',
-	].join(','),
-));
+// Look for the CI details dropdown in the latest 2 days of commits #2990
+const ciDetailsSelector = [
+	'.TimelineItem--condensed:nth-of-type(-n+2) .commit-build-statuses', // TODO[2022-04-29]: GHE #4294
+	'.TimelineItem--condensed:nth-of-type(-n+2) batch-deferred-content[data-url$="checks-statuses-rollups"]',
+].join(',');
 
-async function init(): Promise<false | void> {
-	const icon = await getIcon();
-	if (!icon) {
-		return false;
+async function getCiDetails(): Promise<HTMLElement | undefined> {
+	if (pageDetect.isRepoHome()) {
+		// Clone the CI details if they're already loaded, otherwise clone the include-fragment
+		return select('.file-navigation + .Box :is(.commit-build-statuses, .js-details-container include-fragment[src*="/rollup?"])')!.cloneNode(true);
+	}
+
+	if (pageDetect.isRepoCommitList() && getCurrentCommittish() === await getDefaultBranch()) {
+		return select(ciDetailsSelector)!.cloneNode(true);
 	}
 
+	const dom = await fetchDom(buildRepoURL('commits'));
+	const ciDetails = select(ciDetailsSelector, dom);
+	if (ciDetails && (pageDetect.isDiscussion() || pageDetect.isDiscussionList())) {
+		const style = select('link[href*="/assets/github-"]', dom)!;
+		document.head.append(style); // #5283
+	}
+
+	return ciDetails;
+}
+
+function appendCiDetailsToRepoTitle(ciDetails: HTMLElement): void {
 	// Append to title (aware of forks and private repos)
 	const repoNameHeader = select('[itemprop="name"]')!.parentElement!;
-	repoNameHeader.append(icon);
+	repoNameHeader.append(ciDetails);
```

Comment:
Took the opportunity to do some renaming, since we're not really dealing with an icon in this case.

While we're at it, should we rename the feature `ci-details`? Or even `repo-header-ci-details`?

Reply:
`repo-header-` prefix is fine but I feel "details" could be misunderstood that we are providing more details to CI status, which is not true. "Icon" is more obvious here, just my opinion.